pennylane/tape/qscript.py

```diff
@@ -592,6 +581,8 @@ def trainable_params(self):
         >>> qscript.get_parameters()
         [0.432]
         """
+        if self._trainable_params is None:
+            self._trainable_params = list(range(len(self._par_info)))
         return self._trainable_params
 
     @trainable_params.setter
```

Comment:
should we deprecate this setter, and suggest new tapes be created? maybe in another PR since ~10k tests will fail

Reply:
theoretically nice, but would be really painful.

Reply:
While we could remove it from source code, `.trainable_params = ` occurs 290 times in the tests.